pkgs/applications/networking/irc/halloy/default.nix

```diff
@@ -57,8 +58,25 @@ rustPlatform.buildRustPackage rec {
     wayland
   ];
 
+  desktopItem = makeDesktopItem {
+    name = "org.squidowl.halloy";
+    exec = pname;
+    icon = "org.squidowl.halloy";
+    desktopName = "Halloy";
+    categories = [ "Network" "IRCClient" ];
+    comment = "IRC client written in Rust";
+    keywords = [ "IM" "Chat" ];
+    terminal = false;
+    type = "Application";
+  };
+
+  postInstall = ''
+    install -Dm644 assets/logo_128px.png $out/share/icons/hicolor/128x128/apps/org.squidowl.halloy.png
+    install -Dm644 -t $out/share/applications ${desktopItem}/share/applications/*
+  '';
+
   meta = with lib; {
-    description = "IRC application";
+    description = "IRC client written in Rust, with the Iced GUI library";
```

Comment:
Users don't care in what language a application is written and what kind of libraries are in use.